client/src/components/pages/Home.js

```diff
@@ -36,6 +36,11 @@ class Home extends Component {
 			selectedVideo: videoItems[0]
 		});
 	};
+	handleSave = async video => {
```

Comment:
HandleSave is not receiving any video when it is called on click in videoDetail. You already have the selectedVideo in home, so remove the video parameter and use selectedVideo as the video to be saved.